cli/src/commands/developer/mod.rs

```diff
@@ -0,0 +1,159 @@
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+mod decrypt;
+pub use decrypt::*;
+
+mod deploy;
+pub use deploy::*;
+
+mod execute;
+pub use execute::*;
+
+mod scan;
+pub use scan::*;
+
+use snarkvm::{
+    file::{AleoFile, Manifest},
+    package::Package,
+    prelude::{Program, ProgramID, Transaction},
+};
+
+use anyhow::{bail, ensure, Result};
+use clap::Parser;
+use colored::Colorize;
+use std::{path::PathBuf, str::FromStr};
+
+type CurrentNetwork = snarkvm::prelude::Testnet3;
+
+/// Commands to manage Aleo accounts.
+#[derive(Debug, Parser)]
+pub enum Developer {
+    /// Decrypt a ciphertext
+    Decrypt(Decrypt),
+    /// Deploy a program.
+    Deploy(Deploy),
+    /// Execute a program function.
+    Execute(Execute),
+    /// Scan the node for records.
+    Scan(Scan),
+}
+
+impl Developer {
+    pub fn parse(self) -> Result<String> {
+        match self {
+            Self::Decrypt(decrypt) => decrypt.parse(),
+            Self::Deploy(deploy) => deploy.parse(),
+            Self::Execute(execute) => execute.parse(),
+            Self::Scan(scan) => scan.parse(),
+        }
+    }
+
+    /// Parse the program from the directory.
+    fn parse_program(program_id: ProgramID<CurrentNetwork>, path: Option<String>) -> Result<Program<CurrentNetwork>> {
+        // Instantiate a path to the directory containing the manifest file.
+        let directory = match path {
+            Some(path) => PathBuf::from_str(&path)?,
+            None => std::env::current_dir()?,
+        };
+
+        // Ensure the directory path exists.
+        ensure!(directory.exists(), "The program directory does not exist: {}", directory.display());
+        // Ensure the manifest file exists.
+        ensure!(
+            Manifest::<CurrentNetwork>::exists_at(&directory),
+            "Please ensure that the manifest file exists in the Aleo program directory (missing '{}' at '{}')",
+            Manifest::<CurrentNetwork>::file_name(),
+            directory.display()
+        );
+
+        // Open the manifest file.
+        let manifest = Manifest::<CurrentNetwork>::open(&directory)?;
+        ensure!(
+            manifest.program_id() == &program_id,
+            "The program name in the manifest file does not match the specified program name"
+        );
+
+        // Load the package.
+        let package = Package::open(&directory)?;
+        // Load the main program.
+        let program = package.program();
+        // Prepare the imports directory.
+        let imports_directory = package.imports_directory();
```

Comment:
Eventually we may want to add a flag to download an already deployed program in addition to searching locally.